collector/consumer/exporter/otelexporter/consume.go

```diff
@@ -0,0 +1,96 @@
+package otelexporter
+
+import (
+	"context"
+	"github.com/Kindling-project/kindling/collector/consumer/exporter/otelexporter/defaultadapter"
+	"github.com/Kindling-project/kindling/collector/model"
+	"github.com/Kindling-project/kindling/collector/model/constvalues"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	apitrace "go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+func (e *OtelExporter) Consume(gaugeGroup *model.GaugeGroup) error {
+	if gaugeGroup == nil {
+		// no need consume
+		return nil
+	}
+	gaugeGroupReceiverCounter.Add(context.Background(), 1, attribute.String("name", gaugeGroup.Name))
+	if ce := e.telemetry.Logger.Check(zap.DebugLevel, "exporter receives a gaugeGroup: "); ce != nil {
+		ce.Write(
+			zap.String("gaugeGroup", gaugeGroup.String()),
+		)
+	}
+
+	var hasResult = false
+	for i := 0; i < len(e.adapters); i++ {
+		results, _ := e.adapters[i].Adapt(gaugeGroup)
+		if results != nil && len(results) > 0 {
+			e.Export(results, e.adapters[i])
+			hasResult = true
+		}
+	}
+	if hasResult == false {
+		if ce := e.telemetry.Logger.Check(zap.DebugLevel, "No adapter can deal with this gaugeGroup"); ce != nil {
+			ce.Write(
+				zap.String("gaugeGroup", gaugeGroup.String()),
+			)
+		}
+	}
+	return nil
+}
+
+func (e *OtelExporter) Export(results []*defaultadapter.AdaptedResult, adapter defaultadapter.Adapter) {
+	for i := 0; i < len(results); i++ {
+		result := results[i]
+		switch result.ResultType {
+		case defaultadapter.Metric:
+			e.exportMetric(result, adapter)
+		case defaultadapter.Trace:
+			e.exportTrace(result)
+		default:
+			e.telemetry.Logger.Error("Unexpected ResultType", zap.String("type", string(result.ResultType)))
+		}
+	}
+}
+
+func (e *OtelExporter) exportTrace(result *defaultadapter.AdaptedResult) {
+	if e.defaultTracer != nil && e.cfg.AdapterConfig.NeedTraceAsResourceSpan {
+		_, span := e.defaultTracer.Start(
+			context.Background(),
+			constvalues.SpanInfo,
+			apitrace.WithAttributes(result.Attrs...),
+		)
+		span.End()
+	} else if e.defaultTracer != nil && e.cfg.AdapterConfig.NeedTraceAsResourceSpan {
```

Comment:
The bool expressions of `if` and `else if` are identical. Any typos here? I think a guard clause is better, that is if `e.defaultTracer` is nil, the function 
 return directly.